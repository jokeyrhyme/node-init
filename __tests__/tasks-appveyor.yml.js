/* @flow */
'use strict';

jest.unmock('execa');

const {
  isNeeded,
  lib: { pruneCache, pruneInstall, updateNodeNpm },
} = require('../lib/tasks/appveyor.yml.js');

test('isNeeded()', async () => {
  const result = await isNeeded({ cwd: process.cwd(), scope: '' });
  expect(typeof result).toBe('boolean');
});

test('pruneCache({}, { hasYarn: false })', () => {
  const results = pruneCache({});
  expect(results).toMatchSnapshot();
});

test('pruneInstall({ ... }, { hasYarn: false })', () => {
  const results = pruneInstall(
    {
      install: ['npm install --global yarn'],
    },
    { hasYarn: false }
  );
  expect(results).toMatchSnapshot();
});
test('pruneInstall({ ... }, { hasYarn: true })', () => {
  const results = pruneInstall(
    {
      install: ['npm install --global yarn'],
    },
    { hasYarn: true }
  );
  expect(results).toMatchSnapshot();
});

test('updateNodeNpm({}, { hasYarn: false })', () => {
  const results = updateNodeNpm({}, { hasYarn: false });
  expect(results).toMatchSnapshot();
});
test('updateNodeNpm({}, { hasYarn: true })', () => {
  const results = updateNodeNpm({}, { hasYarn: true });
  expect(results).toMatchSnapshot();
});
