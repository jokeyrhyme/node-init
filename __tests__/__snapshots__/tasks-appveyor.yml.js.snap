// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`pruneCache({}, { hasYarn: false }) 1`] = `Object {}`;

exports[`pruneInstall({ ... }, { hasYarn: false }) 1`] = `Object {}`;

exports[`pruneInstall({ ... }, { hasYarn: true }) 1`] = `
Object {
  "install": Array [
    "npm install --global yarn",
  ],
}
`;

exports[`updateNodeNpm({}, { hasYarn: false }) 1`] = `
Object {
  "install": Array [
    Object {
      "ps": "Install-Product node $env:nodejs_version x64",
    },
    "npm install --global npm-windows-upgrade",
    "npm-windows-upgrade --no-spinner --npm-version latest",
  ],
}
`;

exports[`updateNodeNpm({}, { hasYarn: true }) 1`] = `
Object {
  "install": Array [
    Object {
      "ps": "Install-Product node $env:nodejs_version x64",
    },
    "npm install --global npm-windows-upgrade",
    "npm-windows-upgrade --no-spinner --npm-version latest",
    "npm install --global yarn",
  ],
}
`;
